.github/workflows/test.yml

```diff
@@ -62,3 +63,89 @@ jobs:
 
     - name: Check for dirty tree
       run: ./check-for-dirty-tree
+
+    - name: Generate build number
+      uses: einaregilsson/build-number@v3
+      with:
+        token: ${{ secrets.github_token }}
+
+  prepare-release:
+    needs: ['build']
+    if: startsWith(github.ref, 'refs/tags/')
+    runs-on: ubuntu-latest
+    # env:
+      # Set to force version number, e.g., when no tag exists.
+      # ARC_VERSION: 0.0.0
+    outputs:
+      UPLOAD_URL: ${{ steps.release.outputs.upload_url }}
+      ARC_VERISION: ${{ env.ARC_VERSION }}
+    steps:
+      - name: Get ARC_VERSION from tag
+        if: env.ARC_VERSION == ''
+        run: |
+          echo "ARC_VERSION=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+          echo "version is: ${{ env.ARC_VERSION }}"
+
+      - name: Create GitHub release
+        id: release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+        with:
+          tag_name: ${{ env.ARC_VERSION }}
+          release_name: ${{ env.ARC_VERSION }}
+
+  release:
+    needs: ['prepare-release']
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        build:
+          - linux
+        # - macos
+        include:
+          - build: linux
+            os: ubuntu-18.04
+            target: x86_64-unknown-linux-musl
+        # - build: macos
+        #   os: macos-latest
+        #   target: x86_64-apple-darwin
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v2
+
+    - name: Install packages (Ubuntu)
+      if: matrix.os == 'ubuntu-18.04'
+      run: ./.github/continuous-integration.sh install-ubuntu-packages
+
+  # - name: Install packages (macOS)
+  #   if: matrix.os == 'macos-latest'
+  #   run: ./.github/continuous-integration.sh install-macos-packages
+
+    - name: Build artifacts
+      run: ./build
+
+    - name: Create package
+      shell: bash
+      run: |
+        package="arc-${{needs.prepare-release.outputs.ARC_VERSION}}-${{matrix.target}}"
+
+        mkdir -p "$package"/{bin,share}
+
+        mv README.md "$package"
+        mv arc-mlir/build/llvm-build/bin/{arc,arc-lang,arc-mlir} "$package/bin"
+        mv arc-runtime "$package/share"
```

Comment:
Does the `arc`-script handle this out of the box?

Reply:
Good point, I think we need it to find the `arc-mlir`, `arc-lang` binaries, and the `arc-runtime` source code. How do we do it? Should we check the paths `/usr/local/share/arc-runtime/` and `/usr/local/bin/{arc-mlir,arc-lang}`?

Reply:
Oh, it looks like you fixed it already

https://github.com/cda-group/arc/blob/9b845e73b2fc19ae21b73a5555c1a726975eabcb/arc-mlir/src/tools/arc.in#L165-L178

Reply:
Hmm ok I see, it looks like we need this directory structure:
```
arc/
  bin/
    arc-mlir
    arc-lang
    arc
  share/
    arc-runtime/
    stdlib/
      stdlib.arc
      stdlib.mlir
```
I forgot to add the `stdlib`, I'll do it now

Reply:
> Oh, it looks like you fixed it already

Look at that! My cache is lossy :)